src/account_manager/builder.rs

```diff
@@ -154,53 +156,63 @@ impl AccountManagerBuilder {
                             .coin_type
                             .ok_or(crate::Error::MissingParameter("coin_type (IOTA: 4218, Shimmer: 4219)"))?,
                     };
+
                     (client_options, secret_manager, coin_type)
                 }
                 // If no account manager data exist, we will set it
-                None => {
-                    // Store account manager data in storage
-                    storage_manager.lock().await.save_account_manager_data(&self).await?;
-                    (
-                        self.client_options
-                            .ok_or(crate::Error::MissingParameter("client_options"))?,
-                        self.secret_manager
-                            .ok_or(crate::Error::MissingParameter("secret_manager"))?,
-                        self.coin_type
-                            .ok_or(crate::Error::MissingParameter("coin_type (IOTA: 4218, Shimmer: 4219)"))?,
-                    )
-                }
+                None => (
+                    self.client_options
+                        .clone()
+                        .ok_or(crate::Error::MissingParameter("client_options"))?,
+                    self.secret_manager
+                        .clone()
+                        .ok_or(crate::Error::MissingParameter("secret_manager"))?,
+                    self.coin_type
+                        .ok_or(crate::Error::MissingParameter("coin_type (IOTA: 4218, Shimmer: 4219)"))?,
+                ),
             };
 
-            let client = client_options.clone().finish()?;
+            self.client_options.replace(client_options.clone());
+            self.secret_manager.replace(secret_manager.clone());
+            self.coin_type.replace(coin_type);
```

Comment:
In the None case above these values have to be defined already in self, so actually we only need to set it in the `Some(ref data) => {` case

Reply:
fixed